### DISCUSSION TITLE
Hyperparemetres optimizaton SGT Regressor

### DISCUSSION THREAD

**Discussion by XaviMartiCamarasa** [online-ml/river#1088] [Q&A] [answered]
Hi all
I'm trying to use SGTRegressor, in an nowcasting experiment. For others like LinearRegression I found "optimal parameters", hopefully not overfitting(how could overcome this in river?)..

But for SGTRegressor with the same dataset is almost impossible a good prediction. At this point I do not know if it is because my "ignorance", because I do not find the  optimal parametres or because simpy SGT Regressor does not perform well on Nowcasting..

Any idea will be welcome!

Many  thanks, as usual, for you time

**Comment by smastelini** [comment, marked as answer]
@XaviMartiCamarasa, "there is no free lunch." So we expect no algorithm is good on everything.

However, without some reproducible code is hard to pinpoint the possible root causes or even check if the problem is a bad choice of hyperparameters or a learning bias problem.

**Reply by XaviMartiCamarasa** [reply]
Hi. I was just thinking in loud voice. No coding here.
How could avoid overfitting when doing a regression task?

Regards

**Reply by smastelini** [reply]
This is fairly philosophical, but when you use progressive validation, you can get up-to-date estimates of how your model performs on data it has never seen before. The performance measurements are reliable sources of information.

On a more technical side, regularization is one of the strategies to avoid overfitting. In the specific case of SGTs, they have a regularization parameter that can be used to control the model's complexity.

**Reply by XaviMartiCamarasa** [reply]
Many thanks!
